### ISSUE TITLE
What's the use case?

### ISSUE THREAD

**Issue by dan-klasson** [kolodziejczak-sz/svelte-redux-connect#1]
As Svelte doesn't rerender the whole component, what use case would using `reselect` have?

**Comment by kolodziejczak-sz** [comment]
Reselect is a library designed to:
- improve the implementation of selectors by letting composing them
- make your selectors more efficient thanks to memoize results; a selector is not recomputed unless one of its arguments changes.

That being said, if the redux store updates, every selector used by mounted components will be recomputed (calculating the new stateProps). Then, connect function is checking if the new stateProps are different than the previous one. By default, connect function uses the shallow comparision (every key of prevStatePros === every key of newStateProps). If props have changed, the connect function is calling $set on connected component letting Svelte to handle props change.

And Svelte is checking if props have changed on its own. 
If they have changed Svelte will mark the component as dirty which will cause component update. 
By default connect function has more restricted props comparison, so if connect function decided props have changed, Svelte will follow.

Take a look at this Svelte Component:
```
<script>
	import { afterUpdate } from 'svelte';
	
	let prop = { a: 'any' };
	const foo = () => prop = { a: 'any' };
	
	afterUpdate(() => console.log('afterupdate'));
</script>
{prop.a}
<button on:click={foo}>Click me</button>
```
Every button click will trigger the re-assign and will cause component update because the reference has changed.
Actually there won't be any DOM update because the function which is responsible of update a text node looks like this (https://github.com/sveltejs/svelte/blob/master/src/runtime/internal/dom.ts):

```
export function set_data(text, data) {
	data = '' + data;
	if (text.data !== data) text.data = data;
}
```
Even if there's no DOM update (text.data and data are equal strings in that example), there's still unnecessary computations like scheduling the component update, calling every component lifecycle and trying to update DOM.

If you have selector that returns new reference by every call (like the results of the immutable array functions filter, map...) Svelte will trigger his own update mechanism like this example.

Reselect can avoid this fact: it can return the previous memoized value, meaning there won't be reference change. So there's possibility that connect function won't even call $set on component and trigger props change on Svelte component.

To sum up: reselect can spare components and selectors unnecessary computations.
